packages/knip/src/reporters/codeclimate.ts

```diff
@@ -0,0 +1,127 @@
+import { createHash } from 'node:crypto';
+import type { Entries } from 'type-fest';
+import type { Issue, IssueRecords, IssueSeverity, IssueSymbol, Report, ReporterOptions } from '../types/issues.js';
+import { toRelative } from '../util/path.js';
+import { getTitle } from './util.js';
+
+type CodeClimateSeverity = 'info' | 'minor' | 'major' | 'critical' | 'blocker';
+
+interface CodeClimateEntry {
+  type: 'issue';
+  check_name: string;
+  description: string;
+  categories: string[];
+  location: {
+    path: string;
+    positions: {
+      begin: {
+        line: number;
+        column: number;
+      };
+    };
+  };
+  severity: CodeClimateSeverity;
+  fingerprint: string;
+}
+
+export default async ({ report, issues }: ReporterOptions) => {
+  const entries: CodeClimateEntry[] = [];
+
+  for (const [type, isReportType] of Object.entries(report) as Entries<Report>) {
+    if (!isReportType) {
+      continue;
+    }
+
+    if (type === 'files') {
+      continue;
+    }
+
+    for (const issue of flatten(issues[type] as IssueRecords)) {
+      const { filePath } = issue;
+
+      if (type === 'duplicates' && issue.symbols) {
+        entries.push(
+          ...issue.symbols.map(symbol => ({
+            type: 'issue' as const,
+            check_name: getTitle(type),
+            description: getSymbolDescription({ symbol, parentSymbol: issue.parentSymbol }),
+            categories: ['Duplication'],
+            location: {
+              path: toRelative(filePath),
+              positions: {
+                begin: {
+                  line: symbol.line ?? 0,
+                  column: symbol.col ?? 0,
+                },
+              },
+            },
+            severity: convertSeverity(issue.severity),
+            fingerprint: createFingerprint(filePath, symbol.symbol, symbol.pos),
+          }))
+        );
+      } else {
+        entries.push({
+          type: 'issue' as const,
+          check_name: getTitle(type),
+          description: getIssueDescription(issue),
+          categories: ['Bug Risk'],
```

Comment:
I wonder what's the best way to go about this one. While I agree "bug risk" is best option if we must choose one from https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md#categories, but it doesn't really cut it. But thanks, let's roll with it as it looks we don't really have another option atm.